### PR TITLE
Add method to retrieve oauth2 credentials

### PR DESCRIPTION
**Pull request by LeviHassel**
I think it'd be valuable for users to be able to retrieve their oauth2 credentials after they authenticate with Spotify. In my specific use case, I'd like to use the returned access and refresh tokens to connect to Spotify outside of my app.

Changes:
* Explicitly use the `oauth2` namespace everywhere the package is referenced in-code to make dependencies clearer
* Change the type of `_client` in `SpotifyApiBase` from `http.BaseClient` to `oauth2.Client`, which extends `http.BaseClient` and includes client credentials (this type was already being passed in)
* Add new `getCredentials()` method that returns `SpotifyApiCredentials` object, add test for it
* Copy most variables and methods from `ouath2.Credentials` into `SpotifyApiCredentials`
  * I chose not to have `SpotifyApiCredentials` extend `ouath2.Credentials`, because that would introduce direct references to `oauth2` when users access the object's fields and there were a few variables/methods that didn't seem useful

### CODE REVIEW

**Comment by rinukkusu** [comment]
Also will this work with out our other authentication flow with key and secret?

**Comment by LeviHassel** [comment]
@rinukkusu Yeah, it works with both authentication methods since they both generate the same type of client. However, some properties are null when authenticating only with key/secret, including `refreshToken` and `tokenEndpoint`.

File: lib/src/spotify_credentials.dart [hunk 1]
```diff
@@ -4,10 +4,68 @@
...
+    }
+  }
+
+  /// Whether or not these credentials have expired.
+  ///
+  /// Note that it's possible the credentials will expire shortly after this is
+  /// called. However, since the client's expiration date is kept a few seconds
+  /// earlier than the server's, there should be enough leeway to rely on this.
+  bool get isExpired =>
+      expiration != null && DateTime.now().isAfter(expiration);
```
[the remark this answers is not in the document]
Comment (LeviHassel) [reply lib/src/spotify_credentials.dart:65]: I just tested it and the `expiration` stored in `oauth2.Credentials` is in the user's local time, so comparing it to the user's current local time should work just fine.

[the remark this answers is not in the document]
Comment (LeviHassel) [reply lib/src/spotify_credentials.dart:66]: Also I just learned that [isAfter](https://api.dart.dev/stable/2.7.2/dart-core/DateTime/isAfter.html) is "independent of whether the time is in UTC or in the local time zone".

[the remark this answers is not in the document]
Comment (rinukkusu) [reply lib/src/spotify_credentials.dart:66]: Okay great, it's just easy to shoot yourself in the foot with timezone stuff, thanks for checking!! :)

**Comment by rinukkusu** [comment]
LGTM then!

**Pull request merged**